Automattic/wp-calypso
# Notify user that they were automatically logged in if we land on a page with relevant query parameters

This is related to D15626-code which adds immediate login links feature to wp.com. In case user is known (already logged in on the current device) we ask no additional questions and display no confirmation screen - we just log them in. To avoid confusion and communicate clearly what happened in such case, this PR adds a middleware that displays a notification with an information like "you were just logged in".

The code doesn't require or verify login token, it only relies on simple query parameters. The assumption here is that this change is safe even if someone handcrafts the URL with these params - worst case scenario is that user will see a message that they are now logged in.

Test plan:
1. Navigate to a few different pages and refresh just to make sure things load properly after adding this middleware
2. Add `?logged_via_immediate_link=1` to current URL and press enter
3. Confirm that you saw a notice saying `We logged you in as %(email)s.` with your email in it
4. Confirm that query parameter `logged_via_immediate_link` was removed from the URL
2. Add `?logged_via_immediate_link=1&login_reason=manual_renewal` to current URL and press enter
3. Confirm that you saw a notice saying `We logged you in as %(email)s so you can go ahead and complete your subscription renewal.` with your email in it
4. Confirm that both query parameters were removed from the URL
